balancer_wrapper.go

```diff
@@ -94,8 +94,8 @@ func newCCBalancerWrapper(cc *ClientConn) *ccBalancerWrapper {
 // the underlying balancer.  This is always executed from the serializer, so
 // it is safe to call into the balancer here.
 func (ccb *ccBalancerWrapper) updateClientConnState(ccs *balancer.ClientConnState) error {
-	errCh := make(chan error)
-	ok := ccb.serializer.Schedule(func(ctx context.Context) {
+	errCh := make(chan error, 1)
```

Comment:
This can stay unbounded, and maybe improved a little:

```go
func uCCS() {
	errCh := make(chan error)
	defer close(errCh)
	uccs := ...
	onFailure := func() {}
...
```

Reply:
We need to block on reading from this channel when the `uccs` callback is successfully scheduled on the serializer (because we want to handle state updates inline). But if we don't send something on the channel from the `onFailure` func, the last line `return <-errCh` will block forever, right? This is equivalent, isn't it? https://go.dev/play/p/GHPD4MLhis6

Reply:
Um, yes, I tried to simplify too much, oops.  You can still make the channel unbuffered (sorry for saying "unbounded"), and not defer the close but call close in the onFailure:

https://go.dev/play/p/oqDNkOs_hj7

Reply:
Done.